tests/app/celery/test_provider_tasks.py

```diff
@@ -638,3 +646,18 @@ def test_deliver_push_nonretryable_exception(
     assert rmock.request_history[0].method == 'POST'
     assert rmock.request_history[0].url == url
     assert rmock.request_history[0].json()
+
+
+@patch('app.celery.provider_tasks.current_app.logger.warning')
+@patch('app.celery.provider_tasks.log_and_update_critical_failure')
+def test_handle_delivery_failure_duplication_prevention(mock_log_critical, mock_logger, notify_api):
+    # Ensure a Duplication prevention RuntimeError is handled appropriately
+    notification_id = str(uuid4())
+    try:
+        raise RuntimeError('Duplication prevention - notification.status = sending')
+    except Exception as e:
+        with pytest.raises(NotificationTechnicalFailureException):
+            _handle_delivery_failure(None, None, None, e, notification_id, None)
```

Comment:
I think this works . . .
```suggestion
    e = RuntimeError('Duplication prevention - notification.status = sending')
    with pytest.raises(NotificationTechnicalFailureException):
        _handle_delivery_failure(None, None, None, e, notification_id, None)
```

Reply:
That's how I had it at first, threw errors.